Doc/faq/design.rst

```diff
@@ -528,7 +528,7 @@ is True``) then ``hash(o1) == hash(o2)`` (ie, ``o1.__hash__() == o2.__hash__()``
 regardless of whether the object is in a dictionary or not.  If you fail to meet
 these restrictions dictionaries and other hash based structures will misbehave.
 
-In the case of ListWrapper, whenever the wrapper object is in a dictionary the
+In the case of ``ListWrapper``, whenever the wrapper object is in a dictionary the
```

Comment:
```suggestion
In the case of :class:`!ListWrapper`, whenever the wrapper object is in a dictionary the
```

To be more semantically correct and syntactically consistant

Reply:
What's the difference between `` :class:`!ListWrapper` `` and `` :class:`ListWrapper` ``? I made the change and tried both, but it seemed to render the same.

It also looks like there's no `append` in [stdtypes](https://docs.python.org/3/library/stdtypes.html) as there is for `sort`, and so `` :meth:`~list.append` `` doesn't quite link to anything and just looks like `list.append()`, but I do feel that it's better than the current `` :meth:`append` `` so happy to make the change.

Reply:
Actually, I'll change it to `` :meth:`list.append` `` without the `~` as it renders as `list.append()` compared to just `append()`.

Reply:
> What's the difference between `` :class:`!ListWrapper` `` and `` :class:`ListWrapper` ``? I made the change and tried both, but it seemed to render the same.

As described in the [Sphinx cross-referencing syntax docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-syntax), prefixing the role content with `!` avoids resolving the reference, which (naturally) doesn't resolve (since this isn't a real class documented with the `.. class::` directive and produces a warning), as well as being inefficient.

> I do feel that it's better than the current :meth:`append` so happy to make the change.

Right now the situation with `list` is a little mixed up, as I believe we discussed in another issue. Eventually, if/when things are fixed, that would link to the right place. The alternative would be either linking to something else, or unresolving the reference with `!`.

> Actually, I'll change it to :meth:`list.append` without the ~ as it renders as list.append() compared to just append().

Sure, if you feel the class name adds value—I included the `~` to keep the existing rendering.